formal/promela/src/testbuilder.py

```diff
@@ -143,6 +143,24 @@ def copy(model, codedir, rtems, modfile):
         yaml.dump(model0, file)
 
 
+def get_config(source_dir):
+    config = dict()
+    with open(source_dir + "/testbuilder.yml") as file:
+        global_config = yaml.load(file, Loader=yaml.FullLoader)
+        for key, val in global_config.items():
+            config[key] = val
+    if Path("testbuilder.yml").exists():
+        with open("testbuilder.yml") as file:
+            local_config = yaml.load(file, Loader=yaml.FullLoader)
+            for key, val in local_config.items():
+                config[key] = val
+
+    if {"spin2test", "rtems", "rsb", "simulator", "testyaml", "testcode", "testexe"} - config.keys():
```

Comment:
You compute the missing keys and assign it to a variable before the if-statement.

The if statement checks that variable, and if "true", it can then use it to print a more informative error message - the current one is not helpful enough.